From 62061c763208f322da3890917069260ebb5bc031 Mon Sep 17 00:00:00 2001
From: Nizan Zorea <nzorea@marvell.com>
Date: Sun, 13 Mar 2016 19:04:20 +0200
Subject: [PATCH 0962/1240] fdt_config: mvebu: set fdt_addr env. to currently
 used FDT location

- usage of 'fdt' command (to view/edit) with the currently FTD used by U-Boot
  required to load the FDT to memory by 'fdt_config load' command.
- User can view/edit the currently used FDT:
  by set 'fdt_addr' env. variable to point to the currently used FDT in memory.

Change-Id: I9e4cfd73035d5c525d4ab030484ad4ab3070e480
Signed-off-by: Nizan Zorea <nzorea@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/28206
Tested-by: Star_Automation <star@marvell.com>
Reviewed-by: Haim Boot <hayim@marvell.com>
Reviewed-by: Hanna Hawa <hannah@marvell.com>
Reviewed-by: Wilson Ding <dingwei@marvell.com>
---
 arch/arm/cpu/armv8/armada3700/soc-init.c | 7 +++++++
 arch/arm/cpu/armv8/armada8k/soc.c        | 6 ++++++
 2 files changed, 13 insertions(+)

diff --git a/arch/arm/cpu/armv8/armada3700/soc-init.c b/arch/arm/cpu/armv8/armada3700/soc-init.c
index cc2b4ee..d0f885b 100644
--- a/arch/arm/cpu/armv8/armada3700/soc-init.c
+++ b/arch/arm/cpu/armv8/armada3700/soc-init.c
@@ -25,9 +25,11 @@
 #include <asm/arch/mbus_reg.h>
 #include <asm/arch-mvebu/mbus.h>
 #include <asm/arch-mvebu/pinctl.h>
+#include <asm/arch-mvebu/fdt.h>
 #include <i2c.h>
 #include <libfdt.h>
 #include <asm/arch/boot_mode.h>
+#include <fdt_support.h>
 
 /* IO expander I2C device */
 #define I2C_IO_EXP_ADDR	0x22
@@ -259,6 +261,11 @@ bool mvebu_is_in_recovery_mode(void)
 #ifdef CONFIG_LAST_STAGE_INIT
 int last_stage_init(void)
 {
+#ifdef CONFIG_MULTI_DT_FILE
+	uint8_t *fdt_blob;
+	fdt_blob = cfg_eeprom_get_fdt();
+	set_working_fdt_addr(fdt_blob);
+#endif
 	/* here we switch back to original mode mode by
 	 * writing I2C chip 4c address 0.
 	 */
diff --git a/arch/arm/cpu/armv8/armada8k/soc.c b/arch/arm/cpu/armv8/armada8k/soc.c
index 710db29..f1ac6a5 100644
--- a/arch/arm/cpu/armv8/armada8k/soc.c
+++ b/arch/arm/cpu/armv8/armada8k/soc.c
@@ -29,6 +29,7 @@
 #include <linux/sizes.h>
 #include <netdev.h>
 #include <mvebu_chip_sar.h>
+#include <fdt_support.h>
 
 #define RFU_GLOBAL_SW_RST		(MVEBU_RFU_BASE + 0x84)
 #define RFU_SW_RESET_OFFSET		0
@@ -220,6 +221,11 @@ bool mvebu_is_in_recovery_mode(void)
 #ifdef CONFIG_LAST_STAGE_INIT
 int last_stage_init(void)
 {
+#ifdef CONFIG_MULTI_DT_FILE
+	uint8_t *fdt_blob;
+	fdt_blob = cfg_eeprom_get_fdt();
+	set_working_fdt_addr(fdt_blob);
+#endif
 	return 0;
 }
 #endif
-- 
1.9.1

